geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/QueryPlanner.scala

```diff
@@ -166,6 +168,35 @@ case class QueryPlanner(sft: SimpleFeatureType,
     }
   }
 
+  /**
+   * Standard iterator of simple features
+   */
+  def adaptQuerySizeIterator(accumuloIterator: KVIter, query: Query, decoder: SimpleFeatureDecoder): SFIter = {
+
+    val querySizeIterator = accumuloIterator.map { kv =>
+      decoder.decode(kv.getValue.get)
+    }
+
+    var scannedBytes: Long = 0
+    var scannedRecords: Long = 0
+    var resultBytes: Long = 0
+    var resultRecords: Long = 0
+    while(querySizeIterator.hasNext) { // TODO How do I do this in a Scala-y way?
+      val cur = querySizeIterator.next()
+      scannedBytes += cur.getAttribute(QuerySizeIterator.SCAN_BYTES_ATTRIBUTE).asInstanceOf[Long]
+      scannedRecords += cur.getAttribute(QuerySizeIterator.SCAN_RECORDS_ATTRIBUTE).asInstanceOf[Long]
+      resultBytes += cur.getAttribute(QuerySizeIterator.RESULT_BYTES_ATTRIBUTE).asInstanceOf[Long]
+      resultRecords += cur.getAttribute(QuerySizeIterator.RESULT_RECORDS_ATTRIBUTE).asInstanceOf[Long]
+    }
```

Comment:
i think you need to call close() on the query size iterator once you've exhausted it.